packages/twenty-server/src/engine/core-modules/twenty-config/cache/config-cache.service.ts

```diff
@@ -0,0 +1,133 @@
+import { Injectable, OnModuleDestroy } from '@nestjs/common';
+
+import { CONFIG_VARIABLES_CACHE_SCAVENGE_INTERVAL } from 'src/engine/core-modules/twenty-config/constants/config-variables-cache-scavenge-interval';
+import { CONFIG_VARIABLES_CACHE_TTL } from 'src/engine/core-modules/twenty-config/constants/config-variables-cache-ttl';
+
+import {
+  ConfigCacheEntry,
+  ConfigKey,
+  ConfigKnownMissingEntry,
+  ConfigValue,
+} from './interfaces/config-cache-entry.interface';
+
+@Injectable()
+export class ConfigCacheService implements OnModuleDestroy {
+  private readonly foundConfigValuesCache: Map<
+    ConfigKey,
+    ConfigCacheEntry<ConfigKey>
+  >;
+  private readonly knownMissingKeysCache: Map<
+    ConfigKey,
+    ConfigKnownMissingEntry
+  >;
+  private cacheScavengeInterval: NodeJS.Timeout;
+
+  constructor() {
+    this.foundConfigValuesCache = new Map();
+    this.knownMissingKeysCache = new Map();
+    this.startCacheScavenging();
+  }
+
+  get<T extends ConfigKey>(key: T): ConfigValue<T> | undefined {
+    const entry = this.foundConfigValuesCache.get(key);
```

Comment:
entry naming is not clear, is'nt it a "value"

Reply:
ok got it, entry is value + ttl
I'm challenging the fact that we need a different ttl for each entry. I feel we should just clear the whole cache periodically